### ISSUE TITLE
Different order of Items for data.test_dl.items in Linux and Windows Environment

### ISSUE THREAD

**Issue by rdutta1999** [fastai/fastai#2491]
**Describe the bug**
```data.test_dl.items``` returns the items in different order when tested in both Linux and Windows environments.
[```data``` is an ```ImageDataBunch```,  Fastai version = ```1.0.60``` ]

**Provide your installation details**
For Windows: 
```text
=== Software === 
python        : 3.7.4
fastai        : 1.0.60
fastprogress  : 0.2.2
torch         : 1.4.0
nvidia driver : 442.19
torch cuda    : 10.1 / is available
torch cudnn   : 7501 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 8192MB | GeForce RTX 2060 SUPER

=== Environment === 
platform      : Windows-10-10.0.18362-SP0
conda env     : fastaienv
python        : C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\python.exe
sys.path      : D:\cactus
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\python37.zip
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\DLLs
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv

C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib\site-packages
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib\site-packages\win32
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib\site-packages\win32\lib
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib\site-packages\Pythonwin
C:\Users\RAJDEEP\Anaconda\Anaconda3\envs\fastaienv\lib\site-packages\IPython\extensions
C:\Users\RAJDEEP\.ipython
```
For Linux (Kaggle Kernel)
```text
=== Software === 
python        : 3.6.6
fastai        : 1.0.60
fastprogress  : 0.2.2
torch         : 1.4.0
torch cuda    : 10.0 / is available
torch cudnn   : 7603 / is enabled

=== Hardware === 
torch devices : 1
  - gpu0      : Tesla P100-PCIE-16GB

=== Environment === 
platform      : Linux-4.19.87+-x86_64-with-debian-9.9
distro        : #1 SMP Mon Feb 10 22:35:42 PST 2020
conda env     : Unknown
python        : /opt/conda/bin/python
sys.path      : /kaggle/lib/kagglegym
/kaggle/lib
/kaggle/input/aerial-cactus-identification
/opt/conda/lib/python36.zip
/opt/conda/lib/python3.6
/opt/conda/lib/python3.6/lib-dynload

/root/.local/lib/python3.6/site-packages
/opt/conda/lib/python3.6/site-packages
/opt/conda/lib/python3.6/site-packages/nolearn-0.6.2.dev0-py3.6.egg
/opt/conda/lib/python3.6/site-packages/tabulate-0.8.6-py3.6.egg
/opt/conda/lib/python3.6/site-packages/PyBrain-0.3.3-py3.6.egg
/opt/conda/lib/python3.6/site-packages/python_Levenshtein-0.12.0-py3.6-linux-x86_64.egg
/opt/conda/lib/python3.6/site-packages/pyeconometrics-1.0.2-py3.6.egg
/src/bq-helper
/opt/conda/lib/python3.6/site-packages/IPython/extensions
/root/.ipython
no nvidia-smi is found
```
**To Reproduce**

```
data = ImageDataBunch.from_csv(path = ".",
                               folder = "train/train/",
                               csv_labels = "train.csv",
                               seed = 42,
                               test = "test/test/",
                               bs = 512,
                               ds_tfms = get_transforms(),
                               size = 32,
                               num_workers = 6).normalize(imagenet_stats)
```
Then we train and finetune the model, followed by predicting the Test Data.
```
test_preds, _ = learn.get_preds(DatasetType.Test)
```
In order to get the Filenames, 
```
test_files = data.test_dl.items
test_files
```

**Expected behavior**

It seems that the ```test_files``` for Windows are sorted by the alphabetical order, whereas for Linux, they are random. Performing inference on test_data should take batches of test_data randomly, and I feel that the Linux version is correct while the Windows one isn't.

**Screenshots**

For Windows,
![image](https://user-images.githubusercontent.com/40257378/74509585-e4d98180-4f27-11ea-9500-bea0c400295a.png)

For Linux,
![image](https://user-images.githubusercontent.com/40257378/74509616-f9b61500-4f27-11ea-8121-e4012d5d7c27.png)

**Additional context**

**Comment by sgugger** [comment]
This comes from the different OS going through files differently. 
The factory method won't change so you need to use the data block API to assemble your data and add your test set. There you will have the possibility to give a list of files, that you can then sort (in v2 there is even a function to get that result directly).
